(Please Dont merge) Import model classes

thuthichchoichu opened the pull request.

khoimaianest committed:
Import model classes

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/uncategory/JsonUtils.kt:
```diff
@@ -0,0 +1,120 @@
+package au.com.safetychampion.data.domain.uncategory
```
ko dùng cái này nữa

thuthichchoichu replied:
CustomValueOptionTypeAdapter, cái class này nó đang dùng đến JsonUtils

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/uncategory/ObservableNullableBoolean.java:
```diff
@@ -0,0 +1,39 @@
+package au.com.safetychampion.data.domain.uncategory;
```
các class Observable cũng ko để ở trong data nữa

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/GlbScreenConfig.kt:
```diff
@@ -0,0 +1,27 @@
+package au.com.safetychampion.data.domain.models
```
class này của UI ko dùng

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/GlobalMessage.kt:
```diff
@@ -0,0 +1,10 @@
+package au.com.safetychampion.data.domain.models
```
Của UI

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/MasterCell.kt:
```diff
@@ -0,0 +1,30 @@
+package au.com.safetychampion.data.domain.models
+
```
ko dùng Master Cell sau này viết cái khác

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/MasterCellState.kt:
```diff
@@ -0,0 +1,7 @@
+package au.com.safetychampion.data.domain.models
+
```
Bỏ luôn vì ko dùng master cell

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/OfflineTask.kt:
```diff
@@ -0,0 +1,88 @@
+
+@Entity(indices = [Index("userId")], tableName = "OfflineTasks")
+data class OfflineTask(
+    @PrimaryKey
```
Offline tasks tạm thời bỏ ra

khoimaianest committed:
Added Main Activity.
Updated build.gradle for data module & app module.

khoimaianest committed:
Implemented call(), callAsList()  and related functions in BaseRepository.
Get activetask demo in MainActivity.kt

khoimaianest committed:
Updated errors handle in BaseRepository.kt

khoimaianest committed:
AssignTaskStatusItemUseCase
Added jsonObjectOf()

khoimaianest committed:
call() with optional key item

khoimaianest committed:
Simple UI for testing APIs.
AssignTaskStatusItemUseCase.
AssignManyTasksStatusItemUseCase.

khoimaianest committed:
JsonExtensions: Added String.asT() and String.asListT()

khoimaianest committed:
Updated

ionate12 commented on data/src/main/java/au/com/safetychampion/data/data/common/TaskAPI.kt:
```diff
@@ -1,12 +1,18 @@
     @POST("tasks/list/active")
-    suspend fun active(): Result<APIResponse>
+    suspend fun active(
+        @Body body: JsonObject
```
body pass thử trực tiếp class payload vào. để ko cần phải convert Json nữa.
vd `ActiveTasksPayload`

ionate12 commented on data/src/main/java/au/com/safetychampion/data/data/common/TaskRepository.kt:
```diff
@@ -1,4 +1,11 @@
+import com.google.gson.JsonObject
+
+interface TaskRepository {
+    suspend fun getAllActiveTask(body: JsonObject): Result<List<Task>>
```
pass payload vào thay vì JsonObject. Dùng JsonObject rất tù...

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/extensions/JsonExtensions.kt:
```diff
@@ -0,0 +1,126 @@
+import com.google.gson.JsonObject
+import com.google.gson.reflect.TypeToken
+
+fun <T> JsonElement.asT(tClass: Class<T>): au.com.safetychampion.data.domain.core.Result<T> {
```
đổi tên `asT` ko có nghĩa... có thể đổi qua `itemOrNull()` và ko pass Result nữa.
bỏ param class đi. Xem code a viết mẫu ở dưới. 
``

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/extensions/JsonExtensions.kt:
```diff
@@ -0,0 +1,126 @@
+    }
+}
+
+fun jsonObjectOf(jsonString: String): JsonObject? {
```
những cái util function này sẽ ko cần thiết nữa. Mình ko nên sử dụng JsonObject ở khắp nơi, mà sẽ đi thuần theo Object Oriented.

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/extensions/BaseRepositoryExtensions.kt:
```diff
@@ -0,0 +1,53 @@
+import au.com.safetychampion.data.domain.core.SCError
+import au.com.safetychampion.data.network.APIResponse
+
+internal interface BaseRepositoryExtensions {
```
viết trực tiếp ở BaseRepository luôn ko cần rườm rà interface làm j

thuthichchoichu replied:
Em nghĩ là sau này mình sẽ có nhiều Extension khác nhau, vd như của DataBase... 
Nên viết thành 1 interface code mình nó sẽ theo 1 cái pattern nhất định. Và nếu muốn sử dụng ở đâu đó ngoài repository thì ko cần phải viết lại code.
Thêm 1 cái nữa là code của Base Repository mình nó sẽ gọn

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/core/SCError.kt:
```diff
@@ -1,15 +1,36 @@
+        return "An error has occurred: $code \n $errDescription"
     }
+
+    class Failure(detailsMsg: List<String> = emptyList()) : SCError(
```
class -> data class

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/usecase/assigntaskstatus/AssignManyTasksStatusItemUseCase.kt:
```diff
@@ -0,0 +1,23 @@
+    suspend operator fun invoke(
+        tasks: List<Task>
+    ): Pair<String, Result<List<TaskAssignStatusItem>>> {
+        val mappedTask = tasks.map(TaskPayload::AssignTask)
```
nên viết model riêng ra AssignStatusPayload. Bỏ hết logic xử lý Payload -> JSON mà sẽ pass Payload trực tiếp vào API luôn.

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/usecase/assigntaskstatus/AssignTaskStatusItemUseCase.kt:
```diff
@@ -0,0 +1,31 @@
+    suspend operator fun invoke(
+        task: Task,
+        userId: String? = null,
+        notes: String? = null,
```
cái này với cái AssignTaskStatusMany là một.

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/base/BaseTask.kt:
```diff
@@ -10,12 +9,14 @@ interface BaseTask {
\ No newline at end of file
+}
+
+class BaseModuleImpl(override val _id: String, override val type: String) : BaseModule
```
Impl -> Model

ionate12 committed:
improve json utils func

khoimaianest committed:
Updated repository functions: Using payload classes instead JsonObject.
JsonExtensions functions:
- asT to itemOrNull(), asList to listOrEmpty().
- returns T instead Result<T>, List instead Result<List<T>>.
Removed BaseRepository Interface

khoimaianest committed:
BasePL class: overrideable toJsonElement() and BasePLTypeAdapter class

khoimaianest committed:
Updated BasePL

khoimaianest committed:
Setup Koin:
 - 3 modules: appModules, useCasesModule and commonModule
 Clean up BasePL.
 Deleted GsonUtil, RetrofitClient

khoimaianest committed:
Updated modules.

khoimaianest committed:
MockK instead mockito.
Updated koinInject.

khoimaianest committed:
removed printStackTrace

khoimaianest committed:
Added a modulename TODO

khoimaianest committed:
Merge branch 'feat/SC-2250_base_payload' into feat/SC-2252

khoimaianest committed:
Done the remaining of Task APIs: AssignTaskUseCase + UnAssignTaskUseCase

ionate12 committed:
feat: Token Manager

khoimaianest committed:
Fixed comments

thuthichchoichu committed:
Merge pull request #3 from ionate12/feat/SC-2252

SC-2252: Setup Koin

ionate12 committed:
Merge pull request #2 from ionate12/feat/SC-2250_base_payload

BasePayload

ionate12 committed:
Merge branch 'feat/SCS-2250_1' into feat/token-manager

# Conflicts:
#	app/src/main/java/au/com/safetychampion/di/AppModules.kt
#	data/src/main/java/au/com/safetychampion/util/GsonManager.kt

ionate12 committed:
udp: resolve Conflict and refine token manager

ionate12 committed:
refine DataStore Type

ionate12 committed:
clean up

ionate12 committed:
rename StoreKeys

ionate12 committed:
move classes

ionate12 committed:
improve: TokenManager suspend func

ionate12 committed:
add todo

thuthichchoichu committed:
Merge pull request #4 from ionate12/feat/token-manager

Feat/token manager

khoimaianest committed:
Implement Action APIs

khoimaianest committed:
Small update
